### PR TITLE
The FreeCommAlgebra on a coproduct of types

### PR DESCRIPTION
**Pull request by felixwellen**
This PR is about proving the relation
```
R[I ⊎ J] = R[I][J]
```
where ```I``` and ```J``` are types which function as index-types for the variables in the generalized polynomial-rings ```R[I]``` and ```R[I][J]```. This can be used to show that 
```
R[Fin n][Fin m] = R[Fin (n+m)]
```
which are the standard multivariate polynomials in n,m and n+m variables. This also works for sets merely equivalent to some ```Fin k```, so a version with actions of permutations can also be defined.
The PR also adds base change for commutative algebras (from R-Algebras to S-Algebras along a hom f:S->R).

### CODE REVIEW

**Review by MatthiasHu — CHANGES_REQUESTED** [review]
Great stuff! :-)
Here are some comments.

File: Cubical/Algebra/CommAlgebra/Properties.agda [hunk 1]
```diff
@@ -365,3 +358,63 @@ module _ {R : CommRing ℓ} {A B : CommAlgebra R ℓ} where
...
+      useIso = Iso.leftInv isoToRingHom A
+
+module _ {R S T : CommRing ℓ} (f : CommRingHom S R) (g : CommRingHom T S) where
+  open RingHoms
+  baseChangeComp : (A : CommAlgebra R ℓ)
+                 → baseChange g (baseChange f A) ≡ baseChange (f ∘r g) A
+  baseChangeComp A =
+      compAndBundle (Iso.fun isoS (Iso.inv isoS changedAasHom)) ≡⟨ cong compAndBundle (Iso.rightInv isoS changedAasHom) ⟩
+      Iso.inv isoT (fst asHomA , ((snd asHomA) ∘r f) ∘r g)      ≡⟨ congStep  ⟩
+      baseChange (compRingHom g f) A ∎
```
[the remark this answers is not in the document]
Comment (felixwellen) [reply Cubical/Algebra/CommAlgebra/Properties.agda:408]: I realized, I should remove this part - it is not needed by the PR and too slow to type check.

File: Cubical/Algebra/CommAlgebra/FreeCommAlgebra/Properties.agda [hunk 2]
```diff
@@ -352,6 +352,46 @@ Iso.leftInv (homMapIso {R = R} {I = I} A) =
   λ f → Σ≡Prop (λ f → isPropIsCommAlgebraHom {M = R [ I ]} {N = A} f)
                (Theory.homRetrievable A f)
 
+inducedHomUnique :
+  {R : CommRing ℓ} {I : Type ℓ'} (A : CommAlgebra R ℓ'') (φ : I → fst A )
+  → (f : CommAlgebraHom (R [ I ]) A) → ((i : I) → fst f (Construction.var i) ≡ φ i)
+  → f ≡ inducedHom A φ
```
[the remark this answers is not in the document]
Comment (felixwellen) [reply Cubical/Algebra/CommAlgebra/FreeCommAlgebra/Properties.agda:358]: Yes!

File: Cubical/Algebra/CommAlgebra/FreeCommAlgebra/Properties.agda [hunk 3]
```diff
@@ -352,6 +352,46 @@ Iso.leftInv (homMapIso {R = R} {I = I} A) =
...
+
+{- A corollary, which is useful for constructing isomorphisms to
+   algebras with the same universal property -}
+isIdByUMP : {R : CommRing ℓ} {I : Type ℓ'}
+          → (f : CommAlgebraHom (R [ I ]) (R [ I ]))
+          → ((i : I) → fst f (Construction.var i) ≡ Construction.var i)
+          → (x : ⟨ R [ I ] ⟩) → fst f x ≡ x
+isIdByUMP {R = R} {I = I} f p = equalByUMP (R [ I ]) f (idCAlgHom (R [ I ])) p
+
 homMapPath : {R : CommRing ℓ} {I : Type ℓ'} (A : CommAlgebra R (ℓ-max ℓ ℓ'))
```
[the remark this answers is not in the document]
Comment (felixwellen) [reply Cubical/Algebra/CommAlgebra/FreeCommAlgebra/Properties.agda:395]: Sure...

File: Cubical/Algebra/CommAlgebra/FreeCommAlgebra/Properties.agda [hunk 4]
```diff
@@ -352,6 +352,46 @@ Iso.leftInv (homMapIso {R = R} {I = I} A) =
...
+
+        ind∘ev≡ : inducedHom A (evaluateAt A f) ≡ inducedHom A (evaluateAt A (inducedHom A φ))
+        ind∘ev≡ = cong (inducedHom A) (λ j i → ev≡ i j)
+
+{- Corollary: Two homomorphisms with the same values on generators are equal -}
+equalByUMP : {R : CommRing ℓ} {I : Type ℓ'}
+           → (A : CommAlgebra R ℓ'')
+           → (f g : CommAlgebraHom (R [ I ]) A)
+           → ((i : I) → fst f (Construction.var i) ≡ fst g (Construction.var i))
+           → (x : ⟨ R [ I ] ⟩) → fst f x ≡ fst g x
```
[the remark this answers is not in the document]
Comment (felixwellen) [reply Cubical/Algebra/CommAlgebra/FreeCommAlgebra/Properties.agda:375]: Nice!

**Review by MatthiasHu — APPROVED** [review]
This can be merged now imo!

**Pull request merged**